code/datums/ai/sanity/_sanityloss_controller.dm

```diff
@@ -521,62 +599,151 @@
 	var/justice = applyJustice ? get_modified_attribute_level(H, JUSTICE_ATTRIBUTE) : 0
 	return GetEffectiveItemForce(I, considerRangedAttack, justice) > current_highest_force
 
-/datum/ai_controller/insane/murder/proc/FindEnemies()
-	. = FALSE
+///Gets a list of all living in view and inside containers then checks whether they are an allowed target and returns a list.
+/datum/ai_controller/insane/murder/proc/PossibleEnemies(range = aggro_range)
 	var/mob/living/living_pawn = pawn
 	var/atom/movable/sees_from = pawn
 	if(!isturf(living_pawn.loc))
 		sees_from = pawn.loc
-	var/list/potential_enemies = livinginview(9, sees_from)
+	var/list/potential_enemies = livinginview(range, sees_from)
+	. = list()
 
-	if(!LAZYLEN(potential_enemies)) // We aint see shit!
+	if(!LAZYLEN(potential_enemies))
 		return
 
-	var/list/weighted_list = list()
-	for(var/mob/living/L in potential_enemies) // Oh the CHOICES!
-		if(L == living_pawn)
-			continue
-		if(L.status_flags & GODMODE)
-			continue
-		if(L.stat == DEAD)
-			continue
-		if(living_pawn.see_invisible < L.invisibility)
-			continue
-		if(!isturf(L.loc) && !ismecha(L.loc))
+	for(var/L in potential_enemies)
+		if(!CanTarget(L))
 			continue
-		weighted_list += L
-	for(var/i in weighted_list)
+		. += L
+	return
+
+///Return TRUE if you want thing to be attackable
+/datum/ai_controller/insane/murder/proc/CanTarget(atom/movable/thing)
+	var/mob/living/living_pawn = pawn
+	if(living_pawn.see_invisible < thing.invisibility)
+		return FALSE
+	if(isliving(thing))
+		var/mob/living/living_thing = thing
+		if(living_thing == living_pawn)
+			return FALSE
+		if(living_thing.status_flags & GODMODE)
+			return FALSE
+		if(living_thing.stat == DEAD)
```

Comment:
This could possibly be changed to use an attack stat variable like mobs do. We probably don't need to do that though.